baselines/fedavgm/fedavgm/strategy.py

```diff
@@ -0,0 +1,189 @@
+"""Optionally define a custom strategy.
+
+Needed only when the strategy is not yet implemented in Flower or because you want to
+extend or modify the functionality of an existing strategy.
+"""
+
+from logging import WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+from flwr.common import (
+    FitRes,
+    MetricsAggregationFn,
+    NDArrays,
+    Parameters,
+    Scalar,
+    ndarrays_to_parameters,
+    parameters_to_ndarrays,
+)
+from flwr.common.logger import log
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy import FedAvg
+from flwr.server.strategy.aggregate import aggregate
+
+
+class CustomFedAvgM(FedAvg):
+    """Re-implmentation of FedAvgM."""
```

Comment:
Could you elaborate a bit why you are using a custom `FedAvgM` instead of the one currently available?

Reply:
I improved this section with the new class docstring:

```python
    """Re-implmentation of FedAvgM.

    This implementation of FedAvgM diverges from original (Flwr v1.5.0) implementation.
    Here, the re-implementation introduces the Nesterov Accelerated Gradient (NAG),
    same as reported in the original FedAvgM paper:

    https://arxiv.org/pdf/1909.06335.pdf
    """
```